test-context/src/lib.rs

```diff
@@ -202,10 +203,49 @@ where
     Ok((tokio::task::spawn_blocking(f).await??, digests))
 }
 
+pub trait ContainsSubset {
+    // Returns true if the value is a subset of the receiver.
+    fn contains_subset(&self, value: Value) -> bool;
+    // Returns true if the value a deep subset of the receiver.
+    fn contains_deep_subset(&self, value: Value) -> bool;
```

Comment:
@jcrossley3 did a shallow and a deep version so that the caller can choose how strict the field matching is.

Reply:
Very nice! I might suggest putting the trait, impl and tests in another module, maybe `test-context/src/subset.rs`? `lib.rs` is getting kinda crowded, I think.